src/Magnum/Platform/WindowlessEglApplication.cpp

```diff
@@ -157,12 +157,55 @@ WindowlessEglContext::WindowlessEglContext(const Configuration& configuration, G
                 Debug{} << "Platform::WindowlessEglApplication: found" << count << "EGL devices, choosing device" << configuration.device();
             }
 
-            /* Assuming the same thing won't suddenly start failing when called
-               the second time */
-            Containers::Array<EGLDeviceEXT> devices{configuration.device() + 1};
-            CORRADE_INTERNAL_ASSERT_OUTPUT(eglQueryDevices(configuration.device() + 1, devices, &count));
+            UnsignedInt selectedDevice;
+            Containers::Array<EGLDeviceEXT> devices;
+
+            if (configuration.deviceIsCuda()) {
+                devices = Containers::Array<EGLDeviceEXT>{MAX_DEVICES};
+                CORRADE_INTERNAL_ASSERT_OUTPUT(eglQueryDevices(devices.size(), devices, &count));
+                auto eglQueryDeviceAttribEXT = reinterpret_cast<EGLBoolean (*)(EGLDeviceEXT, EGLint, EGLAttrib*)>(eglGetProcAddress("eglQueryDeviceAttribEXT"));
+                auto eglQueryDeviceStringEXT = reinterpret_cast<const char* (*)(EGLDeviceEXT, EGLint)>(eglGetProcAddress("eglQueryDeviceStringEXT"));
+                
+                if (!eglQueryDeviceAttribEXT || !eglQueryDeviceStringEXT) {
+                    Error e;
+                    e << "Platform::WindowlessEglApplication: CUDA device selection requires 'eglQueryDeviceAttribEXT' method, likely a driver issue";
+                    if(!magnumContext || !(magnumContext->internalFlags() & GL::Context::InternalFlag::GpuValidation))
+                        e << Debug::nospace << "; enable --magnum-gpu-validation to see additional info";
+                    return;
+                }
+
+                for (selectedDevice = 0; selectedDevice < UnsignedInt(count); ++selectedDevice) {
+                    if(magnumContext && (magnumContext->internalFlags() & GL::Context::InternalFlag::GpuValidation))
```

Comment:
```suggestion
                    if(magnumContext && (magnumContext->internalFlags() & GL::Context::InternalFlag::DisplayVerboseInitializationLog))
```

The message is not really coming from KHR_debug or EGL_KHR_debug, this way it would get printed when you run with `--magnum-log verbose` (which already prints other things like device count, when enabled).